### PR TITLE
Feature/add mirror

### PR DESCRIPTION
**Pull request by briancorbin**
moving the [mirror service](https://github.com/mobilecoinofficial/full-service-mirror) to this repo and adding it to CI

todo:

- [x] update test script for mirror service

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ cargo-features = ["resolver"]
 resolver = "2"
 members = [
     "full-service",
+    "mirror",
```
Comment (cxloe) [inline Cargo.toml:7]: All of this seems standalone, but perhaps this should be an optional feature? 🤔

Comment (briancorbin) [reply Cargo.toml:7]: not sure what you mean by this?

Comment (cxloe) [reply Cargo.toml:7]: incase someone wants to build FS and isn't going to use the mirror, but not a big deal since it's probably not much extra overhead

Comment (briancorbin) [reply Cargo.toml:7]: that's what the -p flag is for during build, so if you wanted to just specify full-service you would do cargo build -p mc-full-service

Comment (cxloe) [reply Cargo.toml:7]: O

**Pull request merged**